### PR TITLE
Fix/sequp

### PR DESCRIPTION
**Pull request by jkokosar**

### CODE REVIEW

**Comment by mstajdohar** [comment]
Make sure you test the lines that have changed. Codecov reports tests should be improved.

File: resdk/scripts.py [hunk 1]
```diff
@@ -209,53 +212,57 @@ def parse_annotation_file(annotation_file):
...
 
-            resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor'], 'presample': False})
-            # Setting 'presample': False, moves sample from presample to sample endpoint.
+                experiment_type = EXPERIMENT_TYPE.get(annotations[sample_n]['SEQ_TYPE'].upper(), '')
+                if experiment_type:
+                    presample['descriptor']['geo']['experiment_type'] = experiment_type
 
-        else:
-            print("Error uploading {}".format(fn), file=sys.stderr)
+                resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor'], 'presample': True})
```
Comment (dblenkus) [inline resdk/scripts.py]: @JureZmrzlikar fix this in your commit. You can omit `'presample': False` as it will be set to `True` by default. Don't forget to also remove comment in the line below.

File: resdk/scripts.py [hunk 2]
```diff
@@ -209,53 +212,57 @@ def parse_annotation_file(annotation_file):
...
+                # Setting 'presample': False, moves sample from presample to sample endpoint.
+
+            else:
+                print("Error uploading {}".format(sample_n), file=sys.stderr)
```
Comment (dblenkus) [inline resdk/scripts.py]: Use logger instead of print.

Comment (JureZmrzlikar) [inline resdk/scripts.py]: We need to replace print statements with logger calls in the whole script. We should do that in separate PR.

File: resdk/scripts.py [hunk 3]
```diff
@@ -209,53 +212,57 @@ def parse_annotation_file(annotation_file):
...
-            resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor'], 'presample': False})
-            # Setting 'presample': False, moves sample from presample to sample endpoint.
+                experiment_type = EXPERIMENT_TYPE.get(annotations[sample_n]['SEQ_TYPE'].upper(), '')
+                if experiment_type:
+                    presample['descriptor']['geo']['experiment_type'] = experiment_type
 
-        else:
-            print("Error uploading {}".format(fn), file=sys.stderr)
+                resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor'], 'presample': True})
+                # Setting 'presample': False, moves sample from presample to sample endpoint.
```
Comment (JureZmrzlikar) [inline resdk/scripts.py]: @jkokosar delete this comment.

Comment (dblenkus) [inline resdk/scripts.py]: @JureZmrzlikar delete it in your commit and @jkokosar will rebase on it.

Comment (JureZmrzlikar) [inline resdk/scripts.py]: OK, done.

**Comment by jkokosar** [comment]
@JureZmrzlikar How can we test scripts execution?

**Comment by JureZmrzlikar** [comment]
Well, sequp is just a function in scripts.py... So we would test it like any other function. But since it is extremely large and connects to several external resources it is very difficult to mock all the stuff by which it interacts... I have already started with refactoring to disassemble it to smaller functions which can be tested individually... We will use e2e tests to ensure that things are wired up in the desired way. @mstajdohar @dblenkus do you agree?

File: docs/tutorial.rst [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	      type: basic:string
+	  run:
+	    runtime: polyglot
+	    bash: |
+	      samtools idxstats {{bam.bam.file}} > stats.txt
+	      re-save-file stats stats.txt
+	      CHR=`<my_script>.py stats.txt`
+	      re-save chr "${CHR}"
+
+Save the above code to a file ``<processor_name>.yaml`` using a text editor. **Make sure to modify the name of the Python script** ``<my_script>.py`` **that is referenced in the above code**. **You you will create this** ``.py`` **file in the next step**. Analysis steps can include tools available from the Biolinux-based `docker container`_ (e.g. ``samtools``) or custom scripts (e.g. Python, R) defined by the user. We have designed the above processor ``<processor_name>.yaml`` to use the custom script ``<my_script>.py`` to parse the ``samtools idxstats`` results file and report the chromosome name:
```
Comment (JureZmrzlikar) [inline docs/tutorial.rst]: You you >> You

File: resdk/scripts.py [hunk 5]
```diff
@@ -209,52 +212,65 @@ def parse_annotation_file(annotation_file):
...
+                    'instrument_type': annotations[sample_n].get('INSTRUMENT', None),
+                    'seq_date': annotations[sample_n].get('SEQ_DATE', None),
+                }
+
+            # Paired-end reads
+            if annotations[sample_n]['PAIRED_END'] == 'Y' and annotations[sample_n]['FASTQ_PATH_PAIR']:
+                rw_reads = annotations[sample_n]['FASTQ_PATH_PAIR'].split(',')
+                slug = 'upload-fastq-paired'
+                input_['src1'] = fw_reads
+                input_['src2'] = [os.path.join(genialis_seq_dir, f) for f in rw_reads]
```
Comment (JureZmrzlikar) [inline resdk/scripts.py]: Add also os.path.normath() on top of os.path.join, to avoid invalid paths like "/path/to/here/../here/file.fq"

File: resdk/scripts.py [hunk 6]
```diff
@@ -209,52 +212,65 @@ def parse_annotation_file(annotation_file):
...
+                               input=input_,
+                               descriptor=descriptor,
+                               descriptor_schema=descriptor_schema,
+                               data_name=sample_n)
+
+            if data:
+                for up_file in fn:
+                    uploaded_files.append(up_file)
+
+                presample = resolwe.api.presample.get(data=data.id)[0]
```
Comment (JureZmrzlikar) [inline resdk/scripts.py]: This can be now written as `resolwe.presample.filter(data=data.id)[0]`

File: resdk/scripts.py [hunk 7]
```diff
@@ -209,52 +212,65 @@ def parse_annotation_file(annotation_file):
...
+
+                organism = ORGANISMS.get(annotations[sample_n]['ORGANISM'].upper(), '')
+                if organism:
+                    presample['descriptor']['geo']['organism'] = organism
+
+                experiment_type = EXPERIMENT_TYPE.get(annotations[sample_n]['SEQ_TYPE'].upper(), '')
+                if experiment_type:
+                    presample['descriptor']['geo']['experiment_type'] = experiment_type
+
+                resolwe.api.presample(presample['id']).patch({'descriptor': presample['descriptor']})
```
Comment (JureZmrzlikar) [inline resdk/scripts.py]: This can be now written as `presample.update_descriptor(presample['descriptor'])`

**Comment by JureZmrzlikar** [comment]
I have merged this branch with mine (the one that has tests for sequp). All e2e sequp tests pass, which is a very good sign :-) So, when e2e tests are supported on Travis this can be accepted.

**Comment by mstajdohar** [comment]
I have merged this because we will have to wait about a month for e2e tests.
